### ISSUE TITLE
Add documentation page on supported date formats

### ISSUE THREAD

**Issue by victorlin** [nextstrain/augur#882]
#740 improves the help text for `filter`'s `--min-date`/`--max-date`. Similar documentation should also be provided for metadata dates, which is slightly different (e.g. no support for relative dates). Moving some things over from an old wiki page as a starter:

## Overview

Augur supports a variety of date formats:

- `2018` - Year only
- `2018.23` - Numerical (floating point)
- `2018-03` - Year, month (positive only)
- `2018-03-25` - Year, month, day (ISO 8601 date) (positive only)
- `2018-03-XX`  - Year, month, ambiguous day (positive only)
- `2018-XX-XX` - Year, ambiguous month, ambiguous day (positive only)

Generally, this comes down to flavors of numerical or (potentially incomplete) ISO dates.

## Implementation

Internally, Augur stores dates in numerical format for the following reasons:

1. Pre-historic dates (BC) are not supported by some implementations of ISO date. For example, [Python’s own datetime](https://discuss.python.org/t/bc-date-support/582).
2. During initial implementation, dates needed to be numerical for some uses (e.g. timetree) and it was easier to just convert to numerical and treat them this way across the board.

## Related discussions

- [2021-04-30 on discussion forum: Including ancient genomes with dates before 0](https://discussion.nextstrain.org/t/including-ancient-genomes-with-dates-before-0/516)
- [2021-07-12 on Slack by Cornelius: what date formats are supported by Augur?](https://bedfordlab.slack.com/archives/C0K3GS3J8/p1626110866053900)
- [2022-01-03 on Slack by Victor: why numerical dates?](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1641252577012900)

**Comment by huddlej** [comment]
This is a great idea! We should also consider linking out to examples of ISO 8601 dates, since users may not be familiar with this term (or will not know that they know the associated formats). Linking to the ISO 8601 [calendar dates](https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates) and [durations](https://en.wikipedia.org/wiki/ISO_8601#Durations) sections on wikipedia would be fine for this.

**Comment by victorlin** [comment]
Relatedly, I've done a bunch of date parsing work in #854 (see [dates.py](https://github.com/victorlin/augur/blob/filter/sqlite/augur/dates.py)), but this has yet to be merged.

**Comment by j23414** [comment]
Do we want to also support and document the following formats?
* `YYYY-MM (Day unknown)`
* `YYYY (Month and day unknown)`

Fauna's [format_date](https://github.com/nextstrain/fauna/blob/2428cfd7a72487d99fe149bb0689536bab165e76/vdb/upload.py#L190) function seems to process it, but I assume it will be superseded by augur's version. I could also see dropping any parenthetical strings "\s\(.*) as a pre-processing step and outside the scope.

**Comment by victorlin** [comment]
@j23414 yeah, that would be `2018` and `2018-03` in the issue description examples.

The current support for those isn't in dates.py, but rather a hidden feature of augur filter's subsampling logic which only applies to the metadata `date` column during subsampling:

https://github.com/nextstrain/augur/blob/8014186d35c13d9f131cfa4828c6b7f81932909f/augur/filter.py#L935

As a path to follow, we should aim to support the same date formats across different use cases via functions in dates.py.

**Comment by huddlej** [comment]
One way to think about incomplete dates like `YYYY` and `YYYY-MM` would be to standardize/sanitize these to `YYYY-XX-XX` and `YYYY-MM-XX`, respectively, early in a workflow. This could be part of the proposed work in #860.

**Comment by victorlin** [comment]
Jover's comment https://github.com/nextstrain/augur/pull/1770#discussion_r1992223797 made me realize that there is already a small section on supported date formats: https://docs.nextstrain.org/projects/augur/en/stable/faq/metadata.html#format

I'm updating it in #1770. Given the previous comments, I chose not to document `YYYY-MM`/`YYYY` (i.e. [reduced precision ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Reduced_precision)) even though `YYYY` seems to be somewhat common.
